### ISSUE TITLE
listMetrics failed in each Compartments in profile DEFAULT/

### ISSUE THREAD

**Issue by eshneken** [oracle/oci-grafana-metrics#306]
I installed Grafana (latest version) from the OCI marketplace.  Dynamic group and policy set up and verified.  When attempting to add datasource with instance principal authentication, get the following error:

`listMetrics failed in each Compartments in profile DEFAULT/`

Using OCI credentials works fine and allows a datasource to be created.  

Could this policy autocreated by the stack not be sufficient?

`Allow group grafana-dynamic-group to read all-resources in tenancy |  `

**Comment by mamorett** [comment]
Hi @eshneken . Well, depends also how your dynamic group is setup, stated you are using instance principals in this case. However the policies which are confirmed to work are the following:
allow group grafana to read metrics in tenancy
allow group grafana to read compartments in tenancy
where "grafana" in this case is the name of the dynamic group

**Comment by mamorett** [comment]
yes but I think there is a mismatch here between group and dynamic group. Are you assigning this policy to a group (users) or a dynamic group(instances) ?

**Comment by atsaloli** [comment]
I was getting the same error in Grafana, `listMetrics failed in each Compartments in profile DEFAULT/`

It turned out I had used the wrong OCID in my Dynamic Group definition.

I found this out by installing OCI CLI, setting the env var `export OCI_CLI_AUTH=instance_principal` and running the command:

```
oci monitoring metric-data summarize-metrics-data \
  --compartment-id [insert compartment id here] \
  --namespace oci_computeagent \
  --query-text 'CpuUtilization[1m].mean()'
```

At first I was getting 404 Not Found or Not Authorized. After I corrected the Dynamic Group rules, I started getting back proper answers.

You can use the above command to validate access from your Grafana instance, in terms of it being able to read metrics.
